mobile/app/(app)/change-password/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { ViewStyle } from "react-native";
+import { Screen } from "../../../components/Screen";
+import Header from "../../../components/Header";
+import { Icon } from "../../../components/Icon";
+import { router } from "expo-router";
+import { YStack } from "tamagui";
+import Card from "../../../components/Card";
+import Button from "../../../components/Button";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import { Control, Controller, FieldErrors, FieldValues, set, useForm } from "react-hook-form";
+import React from "react";
+import { Typography } from "../../../components/Typography";
+import FormInput from "../../../components/FormInputs/FormInput";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import API from "../../../services/api";
+import ChangePasswordConfirmation from "./change-password-confirmation";
+
+interface FormData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+const ChangePassowrd = () => {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation("change_password");
+  const [credentialsError, setCredentialsError] = React.useState(false);
+  const [succesfullyChanged, setSuccesfullyChanged] = React.useState(false);
+
+  // Form validation schema
+  const formSchema = z
+    .object({
+      currentPassword: z.string().min(1, t("form.current_password.required")),
+      newPassword: z
+        .string()
+        .min(8, t("form.new_password.helper"))
+        .regex(/^(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9]).*$/, t("form.new_password.pattern")),
+      confirmPassword: z.string(),
+    })
+    .refine((data) => data.newPassword === data.confirmPassword, {
+      message: t("form.confirm_password.no_match"),
+      path: ["confirmPassword"],
+    });
+
+  // React Hook Form setup
+  const { handleSubmit, control, formState } = useForm<FormData>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      currentPassword: "Votemonitor1*",
+      newPassword: "Votemonitor1*",
+      confirmPassword: "Votemonitor1*",
+    },
+  });
+  const { errors } = formState;
+
+  // Submit handler - change password
+  const onSubmit = async (data: FormData) => {
+    try {
+      await API.post("auth/change-password", {
```

Comment:
use a react query mutation for this one, as we have onSucess, onError calbacks and loading state.

Reply:
Done